micro-rdk/src/common/log.rs

```diff
@@ -2,40 +2,187 @@ use crate::{
     google::protobuf::{value::Kind, Struct, Timestamp, Value},
     proto::common::v1::LogEntry,
 };
-use chrono::{DateTime, FixedOffset};
-use std::collections::HashMap;
-
-use super::robot::RobotError;
-
-pub fn config_log_entry(time: DateTime<FixedOffset>, err: Option<RobotError>) -> LogEntry {
-    let secs = time.timestamp();
-    let nanos = time.timestamp_subsec_nanos();
-    let level = match err {
-        Some(_) => "error".to_string(),
-        None => "info".to_string(),
-    };
-    let message = match err {
-        Some(err) => format!("could not create robot from config: {err}"),
-        None => "successfully created robot from config".to_string(),
-    };
-    LogEntry {
-        host: "esp32".to_string(),
-        level,
-        time: Some(Timestamp {
-            seconds: secs,
-            nanos: nanos as i32,
-        }),
-        logger_name: "robot_server".to_string(),
-        message,
-        caller: Some(Struct {
-            fields: HashMap::from([(
-                "Defined".to_string(),
-                Value {
-                    kind: Some(Kind::BoolValue(false)),
-                },
-            )]),
-        }),
-        stack: "".to_string(),
-        fields: vec![],
+use async_lock::{Mutex as AsyncMutex, MutexGuard as AsyncMutexGuard};
+use chrono::Local;
+use lazy_static::lazy_static;
+use ringbuf::{LocalRb, Rb};
+use std::{
+    collections::HashMap,
+    mem::MaybeUninit,
+    time::{Duration, Instant},
+};
+
+use super::app_client::PeriodicAppClientTask;
+
+type LogBufferType = LocalRb<(LogEntry, Instant), Vec<MaybeUninit<(LogEntry, Instant)>>>;
```

Comment:
[non-blocking]: could we maybe make a struct with the relevant comments? the reasoning for storing the time and the reasoning for time and logs appear disjoint enough 

ex
```rs
/// as we cannog guarantee time has been set at every instance, we store each log along an instance of Instant...
struct ViamLogEntry {
    entry: LogEntry,
    time: Instant,
}

impl ViamLogEntry {
  fn new(entry, time) -> Self {...}
  fn to_cstr() -> ... 
}
```